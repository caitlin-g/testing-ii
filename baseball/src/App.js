import React, { Component } from "react";

import "./App.css";

import Dashboard from "./components/Dashboard";
import Display from "./components/Display";

class App extends Component {
  state = {
    strike: 0,
    ball: 0,
    foul: 0,
    hit: 0
  };
  addStrike = () => {
    this.setState({ strike: ++this.state.strike });
  };

  addBall = () => {
    this.setState({ ball: ++this.state.ball });
  };

  addFoul = () => {
    this.setState({ foul: ++this.state.foul });
  };

  addHit = () => {
    this.setState({ hit: ++this.state.hit });
  };

  render() {
    return (
      <div className="App">
        <Dashboard
          addStrike={this.addStrike}
          addBall={this.addBall}
          addFoul={this.addFoul}
          addHit={this.addHit}
        />
        <Display
          strike={this.state.strike}
          ball={this.state.ball}
          foul={this.state.foul}
          hit={this.state.hit}
        />
      </div>
    );
  }
}

export default App;
