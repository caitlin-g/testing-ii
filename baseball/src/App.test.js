import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
// import { render, fireEvent } from "react-testing-library";

import "jest-dom/extend-expect";

import App from "./App";

describe("<App />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it.skip("matches snapshot", () => {
    const tree = renderer.create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("mocking", () => {
  it("is mocking me", () => {
    const mock = jest.fn();

    const actual = mock("smile");

    expect(actual).toBeUndefined();
    expect(mock).toHaveBeenCalled();
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
