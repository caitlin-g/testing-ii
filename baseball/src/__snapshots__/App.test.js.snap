// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Dashboard"
  >
    <h2>
      Baseball Dashboard
    </h2>
    <button
      data-testid="addStrike"
      onClick={[Function]}
    >
      Strike
    </button>
    <button
      data-testid="addBall"
      onClick={[Function]}
    >
      Ball
    </button>
    <button
      onClick={[Function]}
    >
      Foul
    </button>
    <button
      onClick={[Function]}
    >
      Hit
    </button>
  </div>
  <h3>
    Strikes: 
    0
     
  </h3>
  <h3>
    Balls: 
    0
     
  </h3>
  <h3>
    Fouls: 
    0
     
  </h3>
  <h3>
    Hits 
    0
     
  </h3>
</div>
`;
